File: JavaScript/647-palindromic-substrings/palindromic-substrings.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        if len(s) <=1:
            return 1
        n = len(s)
        count = 0
        for i in range(n):
            l = i
            r = i
            while l >= 0 and r < n and s[l] == s[r]:
                l-=1
                r+=1
                count+=1
            l = i
            r = i+1
            while  l >= 0 and r < n and s[l] == s[r]:
                l-=1
                r+=1
                count+=1
        return count

        # for i in range(n):
        #     dp[i][i] = True
        
        # count = 0

        # for length in range(1, n+1):
        #     for i in range(n-length+1):
        #         j = i + length - 1
        #         if s[i] == s[j] and (length <= 2 or  dp[i+1][j-1]):
        #             dp[i][j] = True
        #             count+=1
        # return count
                    

File: JavaScript/647-palindromic-substrings/palindromic-substrings.java
class Solution {
    public int countSubstrings(String s) {
        int out = 0;

        int i = 0;
        while(i < s.length()) {

            int l = i;
            int r = i;

            while(l>=0 && r<s.length() && s.charAt(l) == s.charAt(r)) {
                l-=1;
                r+=1;
                out++;
            }

            l = i;
            r = i+1;

            while(l>=0 && r<s.length() && s.charAt(l) == s.charAt(r)) {
                l-=1;
                r+=1;
                out++;
            }
            i++;


        }
        return out;
        

    }
}